tests/sentry/receivers/test_default_detector.py

```diff
@@ -0,0 +1,214 @@
+
+        project = self.create_project()
+        # Get the team that was auto-created with the project
+        team = project.teams.first()
+        assert team is not None
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            create_metric_detector_with_owner(project, user=self.user)
+
+        detector = Detector.objects.get(project=project, type=MetricIssue.slug)
+        assert detector.name == "High Error Count (Default)"
+        assert detector.owner_team_id == team.id
+        assert detector.enabled is True
+
+    @with_feature(
+        {
+            "organizations:default-anomaly-detector": True,
+            "organizations:anomaly-detection-alerts": False,
+        }
+    )
+    def test_creates_disabled_detector_when_plan_feature_missing(self):
+        """Test that detector is created but disabled when anomaly-detection-alerts is off."""
+        from sentry.receivers.project_detectors import create_metric_detector_with_owner
+
+        project = self.create_project()
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            create_metric_detector_with_owner(project, user=self.user)
+
+        detector = Detector.objects.get(project=project, type=MetricIssue.slug)
+        assert detector.enabled is False
+
+    @with_feature({"organizations:default-anomaly-detector": False})
+    def test_does_not_create_detector_when_feature_disabled(self):
+        """Test that detector is not created when feature flag is disabled."""
+        from sentry.receivers.project_detectors import create_metric_detector_with_owner
+
+        project = self.create_project()
+
+        create_metric_detector_with_owner(project, user=self.user)
+
+        assert not Detector.objects.filter(project=project, type=MetricIssue.slug).exists()
+
+    @with_feature("organizations:default-anomaly-detector")
+    def test_creates_detector_without_team(self):
+        """Test that detector is created even when project has no teams."""
+        from sentry.receivers.project_detectors import create_metric_detector_with_owner
+
+        project = self.create_project()
+        # Remove all teams
+        project.teams.clear()
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            create_metric_detector_with_owner(project, user=self.user)
+
+        detector = Detector.objects.get(project=project, type=MetricIssue.slug)
+        assert detector.owner_team_id is None
+
+
+class TestDisableDefaultDetectorCreation(TestCase):
+    def test_context_manager_disables_signal(self):
+        """Test that disable_default_detector_creation prevents default detectors."""
+        from sentry.grouping.grouptype import ErrorGroupType
+        from sentry.receivers.project_detectors import disable_default_detector_creation
+        from sentry.workflow_engine.typings.grouptype import IssueStreamGroupType
```

Comment:
Are there circular import issues with these, or can they be moved to the top of the file?